images/springdalelinux.yaml

```diff
@@ -202,7 +202,7 @@ files:
 - path: /etc/dracut.conf.d/lxd.conf
   generator: dump
   content: |-
-    add_drivers+=" virtio_scsi virtio_console sd_mod "
```

Comment:
I'm not 100% sure of what dracut does but it looks like it's asking for the virtio-blk driver to be bundled in the initramfs even if not detect at initramfs build time. If that's the case, nice catch!

That said, I'm not entirely sure if we should do that and at least we didn't do it for NVME disks either and I'm not aware of anyone complaining. Using a non-default driver is probably an edge case so maybe it's OK to expect users making that choice to do some customization to the image they want to use?

To be honest though, bundling that driver might be really quick and cheap? I couldn't check the size of the said module on Ubuntu as it's builtin.

@tomponline opinions?

Reply:
> I'm not 100% sure of what dracut does but it looks like it's asking for the virtio-blk driver to be bundled in the initramfs even if not detect at initramfs build time. If that's the case, nice catch!

dracut creates an initramfs image and bundles the specified drivers which aren't included on these distros by default.

> That said, I'm not entirely sure if we should do that and at least we didn't do it for NVME disks either and I'm not aware of anyone complaining. Using a non-default driver is probably an edge case so maybe it's OK to expect users making that choice to do some customization to the image they want to use?

I believe NVMe is included by default for these distros, so we don't need to include it in ```dracut.conf```. I'm not 100% sure on this though.

> To be honest though, bundling that driver might be really quick and cheap? I couldn't check the size of the said module on Ubuntu as it's builtin.

I was able to check the size on one of my VMs, see below for reference:

```lsmod | grep virtio_blk | awk '{print $1, $2/1024 " MB"}'
virtio_blk 20 MB```

After reading through the dracut docs and trying to understand how it works, it appears to be quite efficient at generating a customized initramfs image. With that being said, it doesn't hurt to investigate/test this further.